lib/ProductOpener/APIProductWrite.pm

```diff
@@ -217,6 +281,66 @@ sub update_product_fields ($request_ref, $product_ref) {
 
 			update_field_with_0_or_1_value($request_ref, $product_ref, $field, $value);
 		}
+		# language fields
+		elsif ( ($field =~ /^(.*)_(\w\w)$/)
+			and (defined $language_fields{$1}))
+		{
+
+			my $language_field = $1;
+			my $language_field_lc = $2;
+
+			$request_ref->{updated_product_fields}{$field} = 1;
+
+			$product_ref->{$language_field . '_' . $language_field_lc} = $value;
+		}
+		# tags fields
+		elsif ( ($field =~ /^(.*)_tags(?:_(\w\w))?(_add)?$/)
+			and (defined $writable_tags_fields{$1}))
+		{
+			my $tagtype = $1;
+			# If we are passed a language (e.g. categories_tags_fr, use it
+			# otherwise use the value of the tags_lc request field)
+			my $tags_lc = $2 // $request_body_ref->{tags_lc};
+
+			my $is_addition = $3;
+
+			update_tags_fields($request_ref, $product_ref, $tagtype, $is_addition, $tags_lc, $value);
+		}
+		# Simple product fields
+		elsif (defined $product_simple_fields{$field}) {
+			$product_ref->{$field} = remove_tags_and_quote($value);
+			$request_ref->{updated_product_fields}{$field} = 1;
+		}
+		# Main language
+		elsif ($field eq "lang") {
```

Comment:
Is it a change in strategy from previous API ? I mean, in previous API I could give a lc and it would interpret fields without language using this lc.

What about here ? And what if I change the "lang" of an existing product ? (or is language value aware only on tags fields ?)